src/features/device-state/state-get-utils.ts

```diff
@@ -95,6 +95,14 @@ export function serviceInstallFromImage(
 		is_a_build_of__service: number | { __id: number };
 	},
 ): ExpandedService | undefined;
+export function serviceInstallFromImage(
+	deviceOrFleet:
+		| { service_install: ExpandedServiceInstall[] }
+		| { service: ExpandedService[] },
+	image?: {
+		is_a_build_of__service: number | { __id: number };
+	},
+): ExpandedServiceInstall | ExpandedService | undefined;
 /** @deprecated Use the strongly-typed overloads with `{ service_install }` or `{ service }` instead */
 export function serviceInstallFromImage(
```

Comment:
Still keeping the `AnyObject` overload not to break balena-api (for now), but the follow up might remove it